projects/packages/jetpack-mu-wpcom/src/features/wpcom-block-editor/class-jetpack-wpcom-block-editor.php

```diff
@@ -0,0 +1,630 @@
+		$expected_hash = wp_hash( "$expiration|$action|{$this->nonce_user_id}", 'jetpack_frame_nonce' );
+		remove_filter( 'salt', array( $this, 'filter_salt' ) );
+
+		if ( ! hash_equals( $hash, $expected_hash ) ) {
+			return false;
+		}
+
+		/*
+		 * Failures may return `WP_Error` (showing an error in the iframe) after the
+		 * signature verification passes.
+		 */
+
+		if ( time() > $expiration ) {
+			return new WP_Error( 'nonce_invalid_expired', 'Expired nonce.', array( 'status' => 401 ) );
+		}
+
+		// Check if it matches the current user, unless they're trying to log in.
+		if ( get_current_user_id() !== $this->nonce_user_id && ! doing_action( 'login_init' ) ) {
+			return new WP_Error( 'nonce_invalid_user_mismatch', 'User ID mismatch.', array( 'status' => 401 ) );
+		}
+
+		return true;
+	}
+
+	/**
+	 * Filters the WordPress salt.
+	 *
+	 * @param string $salt Salt for the given scheme.
+	 * @param string $scheme Authentication scheme.
+	 * @return string
+	 */
+	public function filter_salt( $salt, $scheme ) {
+		if ( 'jetpack_frame_nonce' === $scheme ) {
+			$token = ( new Tokens() )->get_access_token( $this->nonce_user_id );
+
+			if ( $token ) {
+				$salt = $token->secret;
+			}
+		}
+
+		return $salt;
+	}
+
+	/**
+	 * Enqueues the WordPress.com block editor integration assets for the editor.
+	 */
+	public function enqueue_block_editor_assets() {
+		global $pagenow;
+
+		// Bail if we're not in the post editor, but on the widget settings screen.
+		if ( is_customize_preview() || 'widgets.php' === $pagenow ) {
+			return;
+		}
+
+		$debug   = defined( 'SCRIPT_DEBUG' ) && SCRIPT_DEBUG;
+		$version = gmdate( 'Ymd' );
+
+		wp_enqueue_script(
+			'wpcom-block-editor-default-editor-script',
+			$debug
+				? '//widgets.wp.com/wpcom-block-editor/default.editor.js?minify=false'
+				: '//widgets.wp.com/wpcom-block-editor/default.editor.min.js',
+			array(
+				'jquery',
+				'lodash',
+				'wp-annotations',
+				'wp-compose',
+				'wp-data',
+				'wp-editor',
+				'wp-element',
+				'wp-rich-text',
+			),
+			$version,
+			true
+		);
+
+		wp_localize_script(
+			'wpcom-block-editor-default-editor-script',
+			'jetpackMuWpcomGutenberg',
+			array(
+				'richTextToolbar' => array(
+					'justify'   => __( 'Justify', 'jetpack-mu-wpcom' ),
+					'underline' => __( 'Underline', 'jetpack-mu-wpcom' ),
+				),
+			)
+		);
+
+		wp_enqueue_script(
+			'wpcom-block-editor-wpcom-editor-script',
+			$debug
+				? '//widgets.wp.com/wpcom-block-editor/wpcom.editor.js?minify=false'
+				: '//widgets.wp.com/wpcom-block-editor/wpcom.editor.min.js',
+			array(
+				'lodash',
+				'wp-blocks',
+				'wp-data',
+				'wp-dom-ready',
+				'wp-plugins',
+			),
+			$version,
+			true
+		);
+		wp_enqueue_style(
+			'wpcom-block-editor-wpcom-editor-styles',
+			$debug
+				? '//widgets.wp.com/wpcom-block-editor/wpcom.editor.css?minify=false'
+				: '//widgets.wp.com/wpcom-block-editor/wpcom.editor.min.css',
+			array(),
+			$version
+		);
+
```

Comment:
This had before an `is_woa_site` I don't think we need it anymore since we moved it away from Jetpack and we want this for both Simple and WoA. Is this correct?

Reply:
Adding `is_woa_site` back in, on a Simple site I still see the widgets.wp.com files being loaded, because it's also enqueued in `gutenberg-wpcom/index.php` (in `gutenberg_wpcom_enqueue_scripts`). Leaving it out doesn't appear to cause any issues, there is still only one lot of assets showing in source. That said it seems to make more sense to leave `is_woa_in` and prevent the additional enqueuing attempts unnecessarily on Simple, I think.

Reply:
You're right, but the idea was to remove duplicities on `gutenberg-wpcom` after this PR. So things that are used in both WoA and simple are only coming from here as suggested here p1716370219731639/1716368140.337279-slack-C05PV073SG3